File: support_tools/swigwin-3.0.12/Examples/test-suite/ocaml/newobject1_runme.ml
open Swig
open Newobject1

exception RuntimeError of string * int

let foo1 = ref (_Foo_makeFoo C_void)
let _ = if get_int (_Foo_fooCount C_void) != 1 then
   raise (RuntimeError ("(1) Foo.fooCount != 1",
			get_int (_Foo_fooCount C_void)))

let foo2 = ref ((invoke !foo1) "makeMore" C_void) 
let _ = if get_int (_Foo_fooCount C_void) != 2 then
   raise (RuntimeError ("(2) Foo.fooCount != 2",
			get_int (_Foo_fooCount C_void)))

let _ = begin
  foo1 := C_void ; Gc.full_major () ;
  (if get_int (_Foo_fooCount C_void) != 1 then
	raise (RuntimeError ("(3) Foo.fooCount != 1",
			     get_int (_Foo_fooCount C_void)))) ;

  foo2 := C_void ; Gc.full_major () ;
  (if get_int (_Foo_fooCount C_void) != 0 then
	raise (RuntimeError ("(4) Foo.fooCount != 0",
			     get_int (_Foo_fooCount C_void)))) ;
end

File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/newobject1_runme.lua
require("import")	-- the import fn
import("newobject1")	-- import code

foo1 = newobject1.Foo_makeFoo()
assert(newobject1.Foo_fooCount() == 1)

foo2 = foo1:makeMore()
assert(newobject1.Foo_fooCount() == 2)

foo1 = nil 
collectgarbage()
assert(newobject1.Foo_fooCount() == 1)

foo2 = nil 
collectgarbage()
assert(newobject1.Foo_fooCount() == 0)
